Lombiq.UIKit/Views/SlickCarousel.cshtml

```diff
@@ -0,0 +1,79 @@
+@using OrchardCore.ContentManagement.Display
+@using OrchardCore.DisplayManagement.ModelBinding
+@using OrchardCore.ResourceManagement.TagHelpers
+@using Microsoft.Extensions.Configuration
+@using OrchardCore.ContentManagement
+
+@inject IContentItemDisplayManager ContentItemDisplayManager
+@inject IUpdateModelAccessor UpdateModelAccessor
+@inject IConfiguration Configuration
+
+<style asp-name="@ResourceNames.SlickTheme"></style>
+<script asp-name="@ResourceNames.Slick" at="Foot"></script>
+
+@{
+    var shapes = Model.Shapes as IEnumerable<dynamic>;
+    if (shapes?.Any() != true) { return; }
+
+    var carouselId = $"carousel-{Guid.NewGuid():N}";
+    var additionalClasses = Model.AdditionalClasses?.ToString() ?? string.Empty;
+}
+
+<div id="@carouselId" class="slickCarousel @additionalClasses">
+    <div class="slickCarousel__carousel">
+        @foreach (var shape in shapes)
+        {
+            <div class="slickCarousel__item">
+                @await DisplayAsync(shape)
+            </div>
+        }
+    </div>
+    <div class="slickCarousel__arrows"></div>
+</div>
+
+<script at="Foot">
+    jQuery(function ($) {
+        const $carouselContainer = $("#@carouselId");
+        const $carousel = $carouselContainer.find(".slickCarousel__carousel");
+        const $carouselArrows = $carouselContainer.find(".slickCarousel__arrows");
+
+        $carousel.on("init", () => {
+            // Carousel initializes elements with empty id attributes, which triggers HTML validation errors.
```

Comment:
Is this always the case? Can't you add ID-s to the elements, or preventing adding the id-s?

If it's always added and there is no way to fill them up (so they will be always empty), then open an issue about this here: https://github.com/kenwheeler/slick

Reply:
@barthamark

Reply:
Done: https://github.com/kenwheeler/slick/issues/4296 I've also added it to the code as a comment.